src/aggregator/aggregator/entry_test.go

```diff
@@ -1071,6 +1073,80 @@ func TestAddUntimed_ResendEnabled(t *testing.T) {
 	require.True(t, ok)
 }
 
+func TestAddUntimed_ClosedAggregation(t *testing.T) {
```

Comment:
this test does a pretty good job illustrating the race condition and leak of the dirty set.